services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/extension/env/EnvExtension.groovy

```diff
@@ -39,10 +39,13 @@ class VirtualEnvExtension extends AbstractGlobalExtension implements SpringConte
 
     @Override
     void notifyContextInitialized(ApplicationContext applicationContext) {
+        applicationContext.autowireCapableBeanFactory.autowireBean(this)
         if (applicationContext.environment.getActiveProfiles().contains("virtual")) {
-            applicationContext.autowireCapableBeanFactory.autowireBean(this)
             buildVirtualEnvironment()
             log.info("Virtual topology successfully created")
+        } else if(applicationContext.environment.getActiveProfiles().contains("hardware")) {
+            labService.createHwLab(topology)
+            log.info("Successfully pointed to a hardware topology")
         }
     }
```

Comment:
It would be nice to add 'else' branch with raising an exception that provided mode is unknown and we should choose one of two modes.

Reply:
done